src/deepagents/sub_agent.py

```diff
@@ -20,7 +20,7 @@ class SubAgent(TypedDict):
 
 def _create_task_tool(tools, instructions, subagents: list[SubAgent], model, state_schema):
     agents = {
-        "general-purpose": create_react_agent(model, prompt=instructions, tools=tools)
+        "general-purpose": create_react_agent(model, prompt=instructions, tools=tools, state_schema=state_schema)
```

Comment:
i had actually not done this because if we need to do this i think we alrso probably want to update other keys? eg if sub agent changes a key, we need to propogate that back up?

Reply:
should be sep pr